packages/x-charts/src/hooks/useTicksGrouped.ts

```diff
@@ -108,97 +102,73 @@ function mapToGrouping(
-    const groupValues = getGrouping(tickValue, dataIndex);
+  for (let groupIndex = 0; groupIndex < groups.length; groupIndex += 1) {
+    for (let dataIndex = 0; dataIndex < tickValues.length; dataIndex += 1) {
+      const tickValue = tickValues[dataIndex];
+      const groupValue = groups[groupIndex].getValue(tickValue, dataIndex);
 
-    groupValues.forEach((groupValue, groupIndex) => {
-      allTickItems.push({
-        value: groupValue,
-        formattedValue: `${groupValue}`,
-        offset: isBandScale(scale)
+      // Check if this is a new unique value for this group
+      const isNew = allTickItems[allTickItems.length - 1]?.value !== groupValue;
+
+      if (isNew) {
+        currentValueCount = 1;
+        // Calculate tick offset
+        const tickOffset = isBandScale(scale)
           ? scale(tickValue)! -
             (scale.step() - scale.bandwidth()) / 2 +
             offsetRatio[tickPlacement] * scale.step()
-          : scale(tickValue),
-        groupIndex,
-        dataIndex,
-        ignoreTick: false,
-        labelOffset: 0,
-      });
-    });
-  });
-
-  // Step 2: Sort by group index to ensure proper ordering
-  allTickItems.sort((a, b) => a.groupIndex - b.groupIndex);
-
-  // Step 3: Process items to create final entries with sync indices and deduplication
-  const entries: GroupedTickItemType[] = [];
-  const dataIndexToLastEntryMap = new Map<number, number>();
-  let currentSyncIndex = -1;
-
-  allTickItems.forEach((item, index) => {
-    const previousItem = allTickItems[index - 1];
-
-    // Check if this starts a new sync group
-    const isNewSyncGroup =
-      index === 0 ||
-      item.value !== previousItem.value ||
-      item.groupIndex !== previousItem.groupIndex;
-
-    if (isNewSyncGroup) {
-      currentSyncIndex += 1;
-
-      // Calculate how many items share this sync index
-      const itemsWithSameSyncIndex = allTickItems.filter((otherItem, otherIndex) => {
-        // Determine the sync index for this other item
-        let otherSyncIndex = -1;
-
-        for (let i = 0; i <= otherIndex; i += 1) {
-          const current = allTickItems[i];
-          const previous = allTickItems[i - 1];
-
-          if (
-            i === 0 ||
-            current.value !== previous.value ||
-            current.groupIndex !== previous.groupIndex
-          ) {
-            otherSyncIndex += 1;
-          }
+          : scale(tickValue);
+
+        // Calculate the label offset
+        const labelOffset =
+          (scale as { step: () => number }).step() *
+          currentValueCount *
+          (offsetRatio[tickLabelPlacement] - offsetRatio[tickPlacement]);
+
+        // Add a new item
+        allTickItems.push({
+          value: groupValue,
+          formattedValue: `${groupValue}`,
+          offset: tickOffset,
+          groupIndex,
+          dataIndex,
+          ignoreTick: false,
+          labelOffset,
+        });
+
+        if (!storedOffsets.has(tickOffset)) {
```

Comment:
Would it be feasible to rely on the `dataIndex` to avoid using floating point number as a key?

This part of the code is quite hard to get. Especially because `storedOffsets` is not very explicit. Whay those offset are stored?

Maybe something like `offsetsToTickIndexes`/`dataIndexToTickIndes` with a comment

```js
// Map the axis dataIndexes to the tick indexes related.
const dataIndexToTicks = new Map<number, Set<number>>();
```

Reply:
We can't use `dataIndex` in this case, we need to know exactly where the tick is, as this is used to remove duplicate ticks.

Floats should be fairly consistent overall, they are just not precise. Can you see an issue that might be created by this?

Reply:
> Can you see an issue that might be created by this?

I don't get why we need the position and not the dataIndex.
The `tickOffset` is computed from the `scale`, the `tickValue=tickValues[dataIndex]`, and `tickPlacement`

Either `tickPlacement` is the same for all groups. In that case, same `tickOffset` is equivalement to same `dataIndex`. Or `tickPlacement` can change from one group to another. But in that case it seems risky to rely on the fact that `tickPlacement='start'` of an item and `tickPlacement='end'` of the previous item get exactly the same floating number